app/components/ui/icon.tsx

```diff
@@ -1,10 +1,10 @@
-// THIS FILE IS GENERATED, edit ./other/svg-icon-template.txt instead
-// then run "npm run build:icons"
-
 import { type SVGProps } from 'react'
 import { cn } from '~/utils/misc.ts'
 import href from './icon.svg'
+
+import type { iconNames } from './icons.json'
 export { href }
+export type IconName = keyof typeof iconNames
```

Comment:
Ok, here me out... What if we make this type somehow resilient to whether there is an `icons.json` file built yet and if it's not it just defaults to `any` or something that will make TypeScript not fail the build if we've not yet generated the icons.json? 🤔

Reply:
The easiest way (only way?) to do that is to add it to the types/deps.d.ts as a module. Typescript will pick it up if it can't find 

https://github.com/epicweb-dev/epic-stack/assets/5633704/10bb9ec7-e600-4c73-b14a-afb662869ab5

Do you think deps.d.ts makes sense or should it be its own definition file?
```ts
declare module '*/icons.json' {
	// This is a fallback for when the /app/components/ui/icons.json file is not generated
	const iconNames: Record<string, string>
	export { iconNames }
}
```

Reply:
Yeah, I think that's alright 👍 good idea!

Reply:
Lots of folks on twitter have been asking why not just generate the type, and the reason is to avoid needing more files with disclaimers not to touch them

But if we generate it to node_modules (like prisma does with .client) or build or somewhere already designated as an output directory, a ts file makes a lot of sense to me

How do you feel about that approach instead?

Reply:
Yes. I definitely prefer that. Would reduce the likelihood of someone accidentally editing the file directly

Reply:
Ok so it looks like node_modules is off limits because any type definition file (required to keep the import from breaking if the file doesn't exist) will override node_modules even if it does exist

Anywhere outside of that works. Top candidates are
- `/build/icon-names.ts`
- `/types/.icon-names.ts`

However, there's no clean way to import from that directory. Our only path aliases are ~ for the app directory and one for the tests directory

I don't love this sort of semi-relative path if it can be avoided

```ts
import type { IconName } from '~/../types/.icon-names.ts'
```

Options are, in my order of preference,
- build directory, accept the weird import
- types directory, accept the weird import
- either of the above with a new path alias for the root directory `@/build/icon-names.ts`
- hidden file in the root of app? `~/.icon-names.ts`